### PR TITLE
Remove dynamic casts, disable RTTI

### PR DESCRIPTION
**Pull request by schaumb**

### CODE REVIEW

File: src/base/anim/controllable.h [hunk 1]
```diff
@@ -16,6 +16,10 @@ class Controllable
 
 	Duration getDuration() const { return duration; };
 
+	virtual std::shared_ptr<void> actualPlot() const {
```
Comment (simzer) [inline src/base/anim/controllable.h:19]: Codes in the base folder planned to be generic helper codes. This class is a generic controllable animation object which should not know about plots. If we need this here, then I suggest to call it data() or some other generic name.

File: src/apps/weblib/jscriptcanvas.h [hunk 2]
```diff
@@ -73,11 +73,15 @@ class JScriptOutputCanvas : public Gfx::ICanvas
 template <class Canvas>
 class OutputCanvasPainterBase :
```
[the remark this answers is not in the document]
Comment (simzer) [reply src/apps/weblib/jscriptcanvas.h:74]: not necessarily in this PR

File: src/base/anim/controllable.h (hunk 1 above)
Comment (schaumb) [reply src/base/anim/controllable.h]: renamed to data

File: src/apps/weblib/jscriptcanvas.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (schaumb) [reply src/apps/weblib/jscriptcanvas.h]: I deleted it

File: project/cmake/common.txt [hunk 3]
```diff
@@ -33,6 +33,7 @@ else()
 		-Wmissing-noreturn \
 		-Wunreachable-code \
 		-Wunused-macros \
+		-fno-rtti \
```
[the remark this answers is not in the document]
Comment (schaumb) [reply project/cmake/common.txt]: OK. 
I'm afraid that this will cause different runtime behavior between desktop and wasm.

**Review by simzer — APPROVED** [review]

**Pull request merged**